tsuru/resource_tsuru_app.go

```diff
@@ -0,0 +1,350 @@
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTsuruApplicationImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Application name",
+				Required:    true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Application description",
+				Optional:    true,
+			},
+			"platform": {
+				Type:        schema.TypeString,
+				Description: "Platform",
+				Required:    true,
+			},
+			"plan": {
+				Type:        schema.TypeString,
+				Description: "Plan",
+				Required:    true,
+			},
+			"team_owner": {
+				Type:        schema.TypeString,
+				Description: "Application owner",
+				Required:    true,
+			},
+			"pool": {
+				Type:        schema.TypeString,
+				Description: "The name of pool",
+				Required:    true,
+			},
+			"tags": {
+				Type:        schema.TypeList,
+				Description: "Tags",
+				Optional:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"metadata": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"labels": {
+							Type:     schema.TypeMap,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"annotations": {
+							Type:     schema.TypeMap,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+			"restart_on_update": {
+				Type:        schema.TypeBool,
+				Description: "Restart app after applying changes",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceTsuruApplicationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	provider := meta.(*tsuruProvider)
+
+	platform := d.Get("platform").(string)
+	if err := validPlatform(ctx, provider, platform); err != nil {
+		return diag.FromErr(err)
+	}
+
+	pool := d.Get("pool").(string)
+	if err := validPool(ctx, provider, pool); err != nil {
+		return diag.FromErr(err)
+	}
+
+	plan := d.Get("plan").(string)
+	if err := validPlan(ctx, provider, plan); err != nil {
+		return diag.FromErr(err)
+	}
+
+	tags := []string{}
+	for _, item := range d.Get("tags").([]interface{}) {
+		tags = append(tags, item.(string))
+	}
+
+	app := tsuru_client.InputApp{
+		Name:      d.Get("name").(string),
+		Platform:  platform,
+		Pool:      pool,
+		Plan:      plan,
+		TeamOwner: d.Get("team_owner").(string),
+		Tags:      tags,
+	}
+
+	if m, ok := d.GetOk("metadata"); ok {
+		metadata := metadataFromResourceData(m)
+		if metadata != nil {
+			app.Metadata = *metadata
+		}
+	}
+
+	if desc, ok := d.GetOk("description"); ok {
+		app.Description = desc.(string)
+	}
+
+	_, resp, err := provider.TsuruClient.AppApi.AppCreate(ctx, app)
+	if err != nil {
+		return diag.Errorf("unable to create app %s: %v", app.Name, err)
+	}
+
+	if resp.StatusCode != http.StatusCreated {
```

Comment:
could we just ignore raw response, in case of errors I believe that `err` will be not nil